### PR TITLE
add flake8 checks

### PR DESCRIPTION
**Pull request by de-code**
bringing it closer to other projects

### CODE REVIEW

File: project_tests.sh [hunk 1]
```diff
@@ -2,5 +2,17 @@
...
-pytest sciencebeam_utils && \
-    pylint sciencebeam_utils
+# avoid issues with .pyc/pyo files when mounting source directory
+export PYTHONOPTIMIZE=
+
+echo "running tests"
+pytest -p no:cacheprovider
+
+echo "running pylint"
+PYLINTHOME=/tmp/sciencebeam-utils \
```
Comment (giorgiosironi) [inline project_tests.sh]: why is this necessary?

Comment (de-code) [reply project_tests.sh]: To be honest it won't be necessary in this case. It was in another project where the Docker user wasn't root. Seems to make sense not to write to the current directory but I could reove that.

**Comment by de-code** [comment]
> The changes forced by flake8 look reasonable, e.g. lambda that was too long

It complaints about any lambda that is assigned to a variable. That is something I would do where a function parameter is optional. I find the rule a bit too much but I am trying to obey.

File: project_tests.sh (hunk 1 above)
Comment (giorgiosironi) [reply project_tests.sh]: It looks like one of those things that either is removed now or will stay there forever; as you like

Comment (de-code) [reply project_tests.sh]: I think I will keep it in. As it would could also cause issues when mounting the source directory and is consistent with other projects. I want `pylint` to only report and don't need output files in the project folder.
